tests/flask_app/test_routes.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+import io
+
+from flask_app.flask_main import create_app
+
+@pytest.fixture
+def client():
+   app = create_app()
+   app.config["TESTING"] = True
+   app.config["UPLOAD_FOLDER"] = 'tests/flask_app/test_data'
+   with app.test_client() as client:
+      yield client
+
+def test_get_upload_page(client):
```

Comment:
What are you referring to when you say "upload_page"? Looking through the different routes in flask_main.py I see an "upload_file" route but not one for page

Reply:
the /data route returns a different result based on the request. There are two request types for the /data route: GET and POST. The GET request returns the HTML page with the "Upload" button, which allows the user to upload the data. So that's one of our "upload pages". The POST request processes the file that was sent (by calling on the generator) and returns the "next actions" page (from flask_app/templates/actions/actions.html). That's the other "upload page". 

I think the names of the test methods I chose were perhaps not the best. I'll rename to: test_get_upload_page and test_uplaod_file.